controllers/vspheremachine_controller.go

```diff
@@ -274,7 +269,37 @@ func (r machineReconciler) reconcileDeleteVMPre7(ctx *context.MachineContext) er
 	return nil
 }
 
+func (r machineReconciler) findVMPre7(ctx *context.MachineContext) (*infrav1.VSphereVM, error) {
+	// Get ready to find the associated VSphereVM resource.
+	vm := &infrav1.VSphereVM{}
+	vmKey := apitypes.NamespacedName{
+		Namespace: ctx.VSphereMachine.Namespace,
+		Name:      ctx.Machine.Name,
+	}
+	// Attempt to find the associated VSphereVM resource.
+	if err := ctx.Client.Get(ctx, vmKey, vm); err != nil {
+		// If an error occurs finding the VSphereVM resource other than
+		// IsNotFound, then return the error. Otherwise it means the VSphereVM
+		// is already deleted, and that's okay.
+		if !apierrors.IsNotFound(err) {
+			return vm, errors.Wrapf(err, "failed to get VSphereVM %s", vmKey)
```

Comment:
Hi @yastij,

You're not relying on `apierrors` to inspect this error any further, right? If you are then wrapping it will cause an inspection to fail.